src/screens/GroupInfo/elements/ModalAction.js

```diff
@@ -13,9 +14,87 @@ const ModalAction = (props) => {
+    from
   } = props;
   const isAndroid = Platform.OS === 'android';
+  const isAnon = selectedUser?.anon_user_info_color_code && selectedUser?.anon_user_info_emoji_code;
+
+  const RenderButton = () => {
+    switch (from) {
+      case ANONYMOUS:
+        if (isAnon) {
+          return (
+            <TouchableOpacity
+              onPress={() => onPress('message', selectedUser)}
+              style={styles.buttonStyle}>
+              <View style={{flexDirection: 'row'}}>
+                <Text style={styles.textButton}>Leave Incognito Mode</Text>
+                <Loading visible={isLoadingInitChat} />
+              </View>
+            </TouchableOpacity>
+          );
+        }
+        return (
+          <>
+            <TouchableOpacity
+              onPress={() => onPress('message-anonymously', selectedUser)}
+              style={styles.buttonStyle}>
+              <Text
+                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
+                Message Incognito
+              </Text>
+            </TouchableOpacity>
+            <TouchableOpacity onPress={() => onPress('view')} style={styles.buttonStyle}>
+              <Text style={styles.textButton}>View {name}&apos;s Profile </Text>
+            </TouchableOpacity>
+          </>
+        );
+      case SIGNED:
+        if (isAnon) {
+          return (
+            <TouchableOpacity
+              onPress={() => onPress('message-anonymously', selectedUser)}
+              style={styles.buttonStyle}>
+              <Text
+                numberOfLines={1}
+                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
+                Message {name} Incognito
+              </Text>
+            </TouchableOpacity>
+          );
+        }
+        return (
+          <>
+            {isGroup && (
+              <TouchableOpacity
+                onPress={() => onPress('message', selectedUser)}
+                style={styles.buttonStyle}>
+                <View style={{flexDirection: 'row'}}>
+                  <Text style={styles.textButton}>Message {name}</Text>
+                  <Loading visible={isLoadingInitChat} />
+                </View>
+              </TouchableOpacity>
+            )}
+            <TouchableOpacity
+              onPress={() => onPress('message-anonymously', selectedUser)}
+              style={styles.buttonStyle}>
+              <Text
+                numberOfLines={1}
+                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
+                Message {name} Incognito
+              </Text>
+            </TouchableOpacity>
+            <TouchableOpacity onPress={() => onPress('view')} style={styles.buttonStyle}>
+              <Text style={styles.textButton}>View {name}&apos;s Profile </Text>
+            </TouchableOpacity>
+          </>
+        );
+      default:
+        <></>;
+    }
+    return <></>;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `RenderButton` function is quite large and complex. It would be better to break it down into smaller, more manageable functions. This will improve readability and maintainability of the code. Here's a suggestion on how you could refactor this:

```diff
-  const RenderButton = () => {
-    switch (from) {
-      case ANONYMOUS:
-        if (isAnon) {
-          return (
-            <TouchableOpacity
-              onPress={() => onPress('message', selectedUser)}
-              style={styles.buttonStyle}>
-              <View style={{flexDirection: 'row'}}>
-                <Text style={styles.textButton}>Leave Incognito Mode</Text>
-                <Loading visible={isLoadingInitChat} />
-              </View>
-            </TouchableOpacity>
-          );
-        }
-        return (
-          <>
-            <TouchableOpacity
-              onPress={() => onPress('message-anonymously', selectedUser)}
-              style={styles.buttonStyle}>
-              <Text
-                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
-                Message Incognito
-              </Text>
-            </TouchableOpacity>
-            <TouchableOpacity onPress={() => onPress('view')} style={styles.buttonStyle}>
-              <Text style={styles.textButton}>View {name}&apos;s Profile </Text>
-            </TouchableOpacity>
-          </>
-        );
-      case SIGNED:
-        if (isAnon) {
-          return (
-            <TouchableOpacity
-              onPress={() => onPress('message-anonymously', selectedUser)}
-              style={styles.buttonStyle}>
-              <Text
-                numberOfLines={1}
-                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
-                Message {name} Incognito
-              </Text>
-            </TouchableOpacity>
-          );
-        }
-        return (
-          <>
-            {isGroup && (
-              <TouchableOpacity
-                onPress={() => onPress('message', selectedUser)}
-                style={styles.buttonStyle}>
-                <View style={{flexDirection: 'row'}}>
-                  <Text style={styles.textButton}>Message {name}</Text>
-                  <Loading visible={isLoadingInitChat} />
-                </View>
-              </TouchableOpacity>
-            )}
-            <TouchableOpacity
-              onPress={() => onPress('message-anonymously', selectedUser)}
-              style={styles.buttonStyle}>
-              <Text
-                numberOfLines={1}
-                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
-                Message {name} Incognito
-              </Text>
-            </TouchableOpacity>
-            <TouchableOpacity onPress={() => onPress('view')} style={styles.buttonStyle}>
-              <Text style={styles.textButton}>View {name}&apos;s Profile </Text>
-            </TouchableOpacity>
-          </>
-        );
-      default:
-        <></>;
-    }
-    return <></>;
-  };
+  const renderAnonymousButton = () => {
+    if (isAnon) {
+      // ...
+    } else {
+      // ...
+    }
+  };
+
+  const renderSignedButton = () => {
+    if (isAnon) {
+      // ...
+    } else {
+      // ...
+    }
+  };
+
+  const RenderButton = () => {
+    switch (from) {
+      case ANONYMOUS:
+        return renderAnonymousButton();
+      case SIGNED:
+        return renderSignedButton();
+      default:
+        return <></>;
+    }
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->